master/pkg/model/experiment_config.go

```diff
@@ -197,6 +226,32 @@ type ResourcesConfig struct {
 	AgentLabel     string  `json:"agent_label"`
 	ResourcePool   string  `json:"resource_pool"`
 	Priority       *int    `json:"priority,omitempty"`
+
+	// omitempty since is not an officially announced feature yet
+	Devices *DevicesConfig `json:"devices,omitempty"`
+}
+
+// ParseJustResources is a helper function for breaking the circular dependency where we need the
+// TaskContainerDefaults to unmarshal an ExperimentConfig, but we need the Resources.ResourcePool
+// setting to know which TaskContainerDefaults to use.  It does not throw errors; if unmarshalling
```

Comment:
I'd rather it throw errors and the caller discard as necessary.

Reply:
Wait... this is still going to be weird because the docstring is basically now, "it always returns a valid ResourcesConfig, even in error cases", which seems equally weird to me.  I actually think it's fine as-is, especially since it's going to go away in a couple of week.